ValveResourceFormat/Resource/ResourceTypes/ModelAnimation/Animation.cs

```diff
@@ -166,14 +164,15 @@ private Frame GetTransformsAtTime(float time)
                 : pDataObject as IKeyValueCollection;
             var frameBlockArray = pData.GetArray("m_frameblockArray");
 
-            FrameCount = pData.GetIntegerProperty("m_nFrames");
-            Frames = new Frame[FrameCount];
+            FrameCount = (int)pData.GetIntegerProperty("m_nFrames");
+            var frameArray = new Frame[FrameCount];
+            Frames = frameArray;
```

Comment:
For the sake of clarity, I would have assigned this after the loop.

Reply:
gotcha. I've made this change now